crypto.go

```diff
@@ -26,13 +28,21 @@ type dhScheme interface {
 	ID() KEMID
 	GenerateKeyPair(rand io.Reader) (KEMPrivateKey, KEMPublicKey, error)
 	Marshal(pk KEMPublicKey) []byte
+	MarshalPrivate(sk KEMPrivateKey) []byte
 	Unmarshal(enc []byte) (KEMPublicKey, error)
+	UnmarshalPrivate(enc []byte) (KEMPrivateKey, error)
 	DH(priv KEMPrivateKey, pub KEMPublicKey) ([]byte, error)
 	PublicKeySize() int
+
+	setEphemeralKeyPair(sk KEMPrivateKey, pk KEMPublicKey)
```

Comment:
I would just take the PrivateKey here (and only store that).  It avoids the risk of mismatch, and means you only have to store one thing.

Reply:
Also, I don't see any calls to this method?

Reply:
Yep, removed!